src/vs/base/test/common/objects.test.ts

```diff
@@ -227,4 +227,59 @@ suite('Objects', () => {
 		assert.strictEqual(obj1.mIxEdCaSe, objects.getCaseInsensitive(obj1, 'MIXEDCASE'));
 		assert.strictEqual(obj1.mIxEdCaSe, objects.getCaseInsensitive(obj1, 'mixedcase'));
 	});
+
+	test('ensureOptionalBooleanValue', () => {
+		const obj = {
```

Comment:
How about testing numbers and string values? Same for the `ensureOptionalNumberValue` test. Ensuring the right type is one of the main reasons we validate these settings, so we better test it aswell.

Reply:
Done